lib/SILOptimizer/Transforms/TempRValueElimination.cpp

```diff
@@ -621,7 +621,7 @@ TempRValueOptPass::tryOptimizeStoreIntoTemp(StoreInst *si) {
     // our base, we fail since those would be re-initializations.
     if (auto *li = dyn_cast<LoadInst>(user)) {
       if (li->getOwnershipQualifier() == LoadOwnershipQualifier::Take) {
-        continue;
+        return {std::next(si->getIterator()), false};
```

Comment:
There is no problem with load[take] in tryOptimizeStoreIntoTemp.
You can keep the optimization here.